resources/js/components/Product/AddToCart.vue

```diff
@@ -171,6 +172,50 @@ export default {
 
             return addition
         },
+
+        getTierPrice(qty) {
+            qty = parseInt(qty)
+            let best = null
+            let bestQty = 0
+            Object.entries(this.tierPrices).forEach(([id, tier]) => {
+                if (tier.qty > bestQty && tier.qty <= qty) {
+                    best = id
+                    bestQty = tier.qty
+                }
+            })
+
+            return best
+        },
+
+        tierPriceDiscount(price, tier) {
+            let discount = tier ?? this.tierPrices[this.tierPrice] ?? null;
+            if(!discount) {
+                return price
+            }
+
+            if (discount.percentage_value > 0) {
+                return price * (100 - discount.percentage_value) / 100
```

Comment:
Just to make sure, the discount should be calculated on the entire price? Not on the price of the qty defined on the tier prices?

e.g. product is 100
Tier price gives a discount of 10% on 5 qty
You order 9
Should the discount be 450 + 400 = 850 or 810?

Reply:
It should be 810, yes. Tier prices like this are not calculated like tax brackets 😉